util/encoding/encoding.go

```diff
@@ -329,12 +329,13 @@ func DecodeUvarint(b []byte) ([]byte, uint64) {
 		panic("insufficient bytes to decode var uint64 int value")
 	}
 	length := int(b[0]) - 8
+	b = b[1:] // skip length byte
 	if length < 0 {
 		panic(fmt.Sprintf("unable to decode negative value into uint64: %d", length))
-	}
-	b = b[1:] // skip length byte
-	if len(b) < length {
-		panic(fmt.Sprintf("insufficient bytes to decode var uint64 int value: %s", b))
+	} else if length > 8 {
```

Comment:
I'd just merge this check with the one above instead of having 2 separate panics: `if length < 0 || length > 8'. The error message for your new panic ("invalid uvarint length of %d") is sufficient to cover both.